multi-agent/adapters/outbound/mongo/blueprint_repository.py

```
@@ -56,19 +51,35 @@ def update(self, *, blueprint_id: str, spec: BlueprintDraft,
                 "updated_at": datetime.now(timezone.utc),
             }}
         )
-
         return res.modified_count == 1
     
     def set_metadata(self, *, blueprint_id: str, metadata: Dict[str, Any]) -> bool:
-        """Set the metadata dictionary for a blueprint document."""
+        """Set individual metadata keys using dot-notation (key-level merge)."""
         if not isinstance(metadata, dict):
             raise ValueError(f"metadata must be a dictionary, got: {type(metadata)}")
```

Ruff (0.15.20)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)